src/foam/web/KeyboardUtil.js

```diff
@@ -0,0 +1,199 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.LIB({
+  name: 'foam.web.KeyboardUtil',
+
+  documentation: 'A utility library to make it easier to work with keyboard events.',
+
+  constants: {
```

Comment:
How does this compare to Element.NAMED_CODES and keyboardShortcuts support?

Reply:
I hadn't seen that stuff before, but I just had a look at it and this utility method seems to be most similar to `Element.evtToCharCode`. The point of it is to provide a reliable way to get a string representing the key a user pressed from a KeyboardEvent. Looks like there are some differences between `Element.evtToCharCode` and this though. `Element.evtToCharCode` uses deprecated properties on KeyboardEvent and uses non-standard strings from the codes, but handles modifier keys like ctrl, alt, and shift and combinations thereof. `KeyboardUtil.getKey` is more standards-compliant and supports more browsers and keys, but doesn't include the modifier keys in the string it returns.

Reply:
Can we unify the two?

Reply:
modifier keys seems like it would be useful to include